architecture/2025-04-compute-platform-extensibility.md

```diff
@@ -0,0 +1,505 @@
+1. Error Handling and Logging: The error handling and logging mechanisms within the Radius codebase will be updated to account for ACI-specific scenarios. This includes handling errors related to ACI resource creation, configuration, and deployment, as well as providing detailed logs for troubleshooting and diagnostics.
+1. Gateways: Radius will provision the necessary Azure resources to establish gateways for routing internet traffic to services hosted in ACI. This includes defining and creating gateways and secret stores for use in ACI containers.
+1. TBD: Secret Management: Radius will provision an Azure KeyVault resource on behalf of the user to store and manage secrets. These secrets can then be referenced in the application definition files to inject secret values as environment variables in ACI containers or to authenticate into services.
+
+<!--
+### Architecture Diagram
+Provide a diagram of the system architecture, illustrating how different
+components interact with each other in the context of this proposal.
+
+Include separate high level architecture diagram and component specific diagrams, wherever appropriate.
+-->
+
+### Detailed Design
+
+#### Key Components and Relationships
+
+1. Azure Container Instances (ACI) Integration
+    * `corerp/renderers/aci` implements ACI-specific resource rendering logic
+    * `corerp/renderers/aci/gateway` provides gateway functionality for ACI
+    * `corerp/renderers/aci/manualscale` handles scaling for ACI containers
+1. Renderer Multiplexer
+    * `corerp/renderers/mux` routes requests to appropriate renderers based on compute environment
+    * Selects between Kubernetes and ACI renderers based on environment configuration
+
+This architecture supports multi-cloud deployments with a unified programming model, where applications can be defined once and deployed to either Kubernetes or Azure Container Instances environments based on the environment configuration.
+
+#### Understanding the Radius/ACI Architecture Relationships
+
+#### Renderer Architecture
+
+The rendering system is a key part of the architecture with several important relationships:
+
+1. Multiplexed Rendering
+    * The `mux.Renderer` acts as a router that directs rendering requests to the appropriate platform-specific renderer
+    * It examines the `EnvironmentComputeKind` (Kubernetes or ACI) from the environment configuration and selects the correct renderer chain
+1. Layered Rendering
+    * Renderers are organized in a layered pattern using composition:
+   KubernetesCompute: kubernetesmetadata → manualscale → daprextension → container
+   ACICompute: aci_manualscale → aci
+    * Each layer handles a specific aspect of rendering, with inner renderers focusing on core functionality and outer renderers adding specialized features
```

Comment:
This is why instead of doing the routing in the router level (routing to aci or kubernetes implementations) we can do it in the renderer level by just checking what the environment type is.

Reply:
After reading the rest, I think that is also what you decided to go with. Happy to discuss further if you'd like.